crates/autopilot/src/run_loop.rs

```diff
@@ -282,6 +282,10 @@ impl RunLoop {
         let solutions = winner_selection.filter_unfair_solutions(solutions, &auction);
         let solutions = winner_selection.mark_winners(solutions);
 
+        // Count and record the number of winners
+        let num_winners = solutions.iter().filter(|p| p.is_winner()).count();
+        Metrics::get().auction_winners.observe(num_winners as f64);
```

Comment:
Let's always use a safer and more idiomatic approach.
```suggestion
        Metrics::get().auction_winners.observe(f64::from(num_winners));
```

Reply:
Could you please elaborate why this is a more idiomatic approach? In this case `num_winners` is an `usize` so the `from`/`try_from` is not implemented for `f64`, unless I am missing something here.

The only other option I see  is to use something like `num_winners.to_f64().expect(...` but (a) `num_winners` should always be a low number and (b) even if for some reason it fails we would not want to panic for a metric

Reply:
`var as something` will always compile even if the var type is changed for some reason, while using from/try_from will fail on the compilation step.

Reply:
if `from` is not implemented, anything other than `as` should be used.

Reply:
Got it. Upgraded to use `num_winners.to_f64().expect(... ` and avoid panic (only log) as it's a metric